src/executors/collect_battery.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Created by xiaofei.lan@rokid.com on 2019/04/26.
+ */
+
+#include "collect_battery.h"
+
+static void readFile(const char *path, char *buf, int *value) {
+  FILE *file;
+  int size = 9;
+  file = fopen(path, "rb");
+  if (file == NULL)
+  {
+    YODA_SIXSIX_FLOG("No such file: %s. skip collect info.", path);
+    if (value != NULL)
+    {
+      *value = 0;
+    }
+    return;
+  }
+  size = fread(buf, 1, size, file);
+  fclose(file);
+  if (size <= 0)
+  {
+    if (value != NULL)
+    {
+    *value = 0;
+    }
+    return;
+  }
+  if (value != NULL) {
+    *value = atoi(buf);
+  }
+}
+
+YODA_NS_BEGIN
+
+CollectBattery::CollectBattery() : IJobExecutor("CollectBattery"),
+                                   _workReq(nullptr) {
+
+}
+
+CollectBattery::~CollectBattery() {
+  YODA_SIXSIX_FASSERT(_workReq == nullptr, "%s uv_work_t* _workReq not nullptr", _name.c_str());
+}
+
+void CollectBattery::execute() {
+  YODA_SIXSIX_SASSERT(!_workReq, "CollectBattery is running");
+  _workReq = new uv_work_t;
+  UV_CB_WRAP1(_workReq, cb1, CollectBattery, doCollect, uv_work_t);
+  UV_CB_WRAP2(_workReq, cb2, CollectBattery, afterCollect, uv_work_t, int);
+  uv_queue_work(uv_default_loop(), _workReq, cb1, cb2);
+}
+
+void CollectBattery::doCollect(uv_work_t *req) {
+  YODA_SIXSIX_SLOG("========== CollectBattery startup  ==========");
+  char buffer[10];
+  memset(buffer, 0, sizeof buffer);
+  timestamp = time(nullptr);
+
+  readFile(GET_CHARGER_FILE("/temp"), buffer, &bat_temp);
+
+  readFile(GET_CHARGER_FILE("/constant_charge_current"), buffer, &current);
+
+  readFile(GET_CHARGER_FILE("/voltage_now"), buffer, &usb_voltage);
+
+  readFile(GET_CHARGER_FILE("/status"), buffer, NULL);
+  strcpy(status, buffer);
```

Comment:
Where is declared?

Reply:
status is a private variable. I have already named start with _.